index.bs

```diff
@@ -355,6 +371,53 @@ or failure:
 	append a duplicate of the second element to |tokens|.
 7. Return |tokens|.
 
+The <dfn for="IntersectionObserver">scrollable intersection rectangles</dfn>
+for an {{IntersectionObserver}}
+are the rectangles of scrollable containers on the path from the
+<a>intersection root</a> to <a for="IntersectionObserver">target</a>
+which could clip the target's rectangle.
+
+When calculating the <a>scrollable intersection rectangles</a> for
+a <a>same-origin-domain target</a>, the rectangle is then expanded
+according to the offsets in the {{IntersectionObserver}}’s {{[[scrollMargin]]}} slot
+in a manner similar to CSS's 'margin' property,
+with the four values indicating the amount the top, right, bottom, and left edges, respectively, are offset by,
+with positive lengths indicating an outward offset.
+Percentages are resolved relative to the width of the undilated rectangle.
+
+Note: {{IntersectionObserver/scrollMargin}} affects the clipping of <a for="IntersectionObserver">target</a>
+by all scrollable ancestors up to and including the <a>intersection root</a>.
```

Comment:
Please state explicitly what the behavior will be for a scrollable `root` if both `scrollMargin` and `rootMargin` are present. Probably should have a test for this as well, if you don't already.

Reply:
Done for the definition. Will add a test to the wpt CL.

Reply:
One of the goals of this new API is to enable lazy loading of images inside image carousels, both by javascript developers today, and with native lazy loading in the future. When this was originally discussed [here](https://github.com/w3c/IntersectionObserver/issues/431#issuecomment-719792663), music.apple.com developers wanted different margins for the horizontal carousel vs the root margin. I think the intention is that `scrollMargin` does not apply to the `rootMargin`. For example, consider the following case with a lazy loaded image next to a carousel:
```
<div id="spacer" style="height: 200vh;"></div>
<img id="lazyimg" loading=lazy style="width: 100px; height: 100px;">
<div id="carousel" style="width: 100px; height: 100px; overflow: hidden; display: inline-block;">
  <img id="lazycarouselimg" loading=lazy style="width: 100px; height: 100px;">
</div>
```
Wouldn't we want `#lazyimg` and `#lazycarouselimg` to start loading at the same time? With `scrollMargin` applying to the `rootMargin`, `#lazycarouselimg` would start loading first.

Reply:
Ah, I think I understand. In the above example, both images would start loading at the same time because both calculate root intersection considering both rootMargin and scrollMargin.

Related question: native lazy loading uses an intersection observer with implementation-defined `rootMargin` values (Chromium's are [here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/frame/settings.json5;drc=844e1228798e3f0571bb3034308d92916bb987ec;l=970)). If native lazy loading starts using a default `scrollMargin` of, say, 500px, will that change the intersection behavior of the common-case of a lazy-loaded image on a regular scrollable page? In other words, will images start loading 500px sooner?

Reply:
Yes, it would, as both of them would apply. For lazy loaded images we could the intersection observer to only use a scroll-margin if it proves to be an issue.
Here's a test case of both scroll and root margin being applied:
https://chromium-review.googlesource.com/c/chromium/src/+/4740913/7/third_party/blink/web_tests/external/wpt/intersection-observer/scroll-and-root-margin.html

Reply:
The music.apple.com developers wanted to use a margin of 200vh for the vertical margin and 50vw for the horizontal margin. We may want to do something similar for native lazy loading, since vertical scrolls are both faster and more common. Would switching the native lazy loading observer from `rootMargin: margin, margin, margin, margin` to `scrollMargin: margin, 0.25*margin, margin, 0.25*margin` achieve this? I think it would, and so I agree with your current approach of applying scrollMargin to the root, which is also elegant from a consistency point of view.

Reply:
Yes, I think it would achieve the desired behavior.